File: capitulo-02-auth/java/GraphAuthExample.java
package capitulo02;

import com.azure.identity.ClientSecretCredential;
import com.azure.identity.ClientSecretCredentialBuilder;
import com.microsoft.graph.serviceclient.GraphServiceClient;

/**
 * Capítulo 2: Autenticación
 * Client Credentials Flow
 */
public class GraphAuthExample {

    public static void main(String[] args) {
        System.out.println("=== Capítulo 2: Autenticación ===\n");

        String tenantId = System.getenv("SP_TENANT_ID");
        String clientId = System.getenv("SP_CLIENT_ID");
        String clientSecret = System.getenv("SP_CLIENT_SECRET");

        if (tenantId == null || clientId == null || clientSecret == null) {
            System.err.println("Error: Faltan variables de entorno");
            return;
        }

        try {
            ClientSecretCredential credential = new ClientSecretCredentialBuilder()
                .tenantId(tenantId)
                .clientId(clientId)
                .clientSecret(clientSecret)
                .build();

            GraphServiceClient client = new GraphServiceClient(credential);
            var org = client.organization().get();

            System.out.println("✓ Conexión exitosa");
            System.out.println("  Tenant: " + org.getValue().get(0).getDisplayName());

        } catch (Exception e) {
            System.err.println("✗ Error: " + e.getMessage());
        }
    }
}

File: capitulo-02-auth/csharp/GraphAuthExample.cs
using Azure.Identity;
using Microsoft.Graph;

namespace Capitulo02Auth
{
    /// <summary>
    /// Ejemplo de autenticación con Microsoft Graph usando Client Credentials
    /// Capítulo 2: La Experiencia SharePoint
    /// </summary>
    public class GraphAuthExample
    {
        public static async Task Main(string[] args)
        {
            Console.WriteLine("=== Capítulo 2: Autenticación ===\n");

            // Configuración desde variables de entorno
            string tenantId = Environment.GetEnvironmentVariable("SP_TENANT_ID")!;
            string clientId = Environment.GetEnvironmentVariable("SP_CLIENT_ID")!;
            string clientSecret = Environment.GetEnvironmentVariable("SP_CLIENT_SECRET")!;

            if (string.IsNullOrEmpty(tenantId) || string.IsNullOrEmpty(clientId) || string.IsNullOrEmpty(clientSecret))
            {
                Console.WriteLine("Error: Faltan variables de entorno.");
                Console.WriteLine("Asegúrate de configurar: SP_TENANT_ID, SP_CLIENT_ID, SP_CLIENT_SECRET");
                return;
            }

            try
            {
                // Crear credencial de cliente
                var credential = new ClientSecretCredential(tenantId, clientId, clientSecret);

                // Crear cliente de Graph
                var graphClient = new GraphServiceClient(credential);

                // Obtener información de la organización
                var org = await graphClient.Organization.GetAsync();

                Console.WriteLine("✓ Conexión exitosa a Microsoft Graph");
                Console.WriteLine($"  Tenant: {org?.Value?.FirstOrDefault()?.DisplayName}");
                Console.WriteLine($"  ID: {org?.Value?.FirstOrDefault()?.Id}");
            }
            catch (Exception ex)
            {
                Console.WriteLine($"✗ Error de autenticación: {ex.Message}");
            }
        }
    }
}
